shared/macros-bash.jinja

```diff
@@ -444,31 +444,31 @@ printf '%s\n' "{{{ message | replace('"', '\\"') }}}" >&2
 # separator_regex: regular expression that describes the separator and surrounding whitespace
 # prefix_regex: regular expression describing allowed leading characters at each line
 #}}
-{{%- macro set_config_file(path, parameter, value, create, insert_after, insert_before, insensitive=true, separator=" ", separator_regex="\s\+", prefix_regex="^\s*") -%}}
+
+{{%- macro set_config_file(path, parameter, value, create, insert_after, insert_before, insensitive=true, separator=" ", separator_regex="\s\+", prefix_regex="^\s*", sed_path_separator="/") -%}}
     {{%- set line_regex = prefix_regex + ((parameter | escape_regex) | replace("/", "\/")) + separator_regex -%}}
-    {{%- set new_line = parameter+separator+value -%}}
```

Comment:
by removing this, `new_line` in line 459 is empty and the remediation is wrongly built

in the remediation there is this empty printf: `printf '%s\n' "" >> "/etc/rsyslog.conf"`

Reply:
Also, I noticed that the remediation writes to `/etc/rsyslog.conf`, meanwhile the test `linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/tests/rsyslogd_wrong_value.fail.sh` writes to `/etc/rsyslog.d/encrypt.conf` and this value is not being removed or updated by the remediation. I wonder what would happen if there would be two config files containing the parameter, my guess is that the `.d` directories take precedence.